Dockerfile

```diff
@@ -1,31 +1,28 @@
 # ---------------------------------------
 # Base image for node
-FROM node:19 as node_base
+FROM node:19-slim as node_base
 
 WORKDIR /usr/src/app
 
 # ---------------------------------------
 # Base image for runtime
-FROM python:3.11 as base
```

Comment:
This commit seem to break local docker builds.
If checking out current master commit ( 428231e65057feeea8005420d1b90f6b42d4a206 )

And trying to run with `docker compose up --build` you will get with this error message:
```
serge-serge-1  | Collecting llama-cpp-python
serge-serge-1  |   Downloading llama_cpp_python-0.1.38.tar.gz (1.1 MB)
serge-serge-1  |      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 23.6 MB/s eta 0:00:00
serge-serge-1  |   Installing build dependencies: started
serge-serge-1  |   Installing build dependencies: finished with status 'done'
serge-serge-1  |   Getting requirements to build wheel: started
serge-serge-1  |   Getting requirements to build wheel: finished with status 'done'
serge-serge-1  |   Preparing metadata (pyproject.toml): started
serge-serge-1  |   Preparing metadata (pyproject.toml): finished with status 'done'
serge-serge-1  | Collecting typing-extensions>=4.5.0 (from llama-cpp-python)
serge-serge-1  |   Downloading typing_extensions-4.5.0-py3-none-any.whl (27 kB)
serge-serge-1  | Building wheels for collected packages: llama-cpp-python
serge-serge-1  |   Building wheel for llama-cpp-python (pyproject.toml): started
serge-serge-1  |   Building wheel for llama-cpp-python (pyproject.toml): finished with status 'error'
serge-serge-1  |   error: subprocess-exited-with-error
serge-serge-1  |   
serge-serge-1  |   × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
serge-serge-1  |   │ exit code: 1
serge-serge-1  |   ╰─> [94 lines of output]
serge-serge-1  |       
serge-serge-1  |       
serge-serge-1  |       --------------------------------------------------------------------------------
serge-serge-1  |       -- Trying 'Ninja' generator
serge-serge-1  |       --------------------------------
serge-serge-1  |       ---------------------------
serge-serge-1  |       ----------------------
serge-serge-1  |       -----------------
serge-serge-1  |       ------------
serge-serge-1  |       -------
serge-serge-1  |       --
serge-serge-1  |       Not searching for unused variables given on the command line.
serge-serge-1  |       -- The C compiler identification is GNU 10.2.1
serge-serge-1  |       -- Detecting C compiler ABI info
serge-serge-1  |       -- Detecting C compiler ABI info - done
serge-serge-1  |       -- Check for working C compiler: /usr/bin/cc - skipped
serge-serge-1  |       -- Detecting C compile features
serge-serge-1  |       -- Detecting C compile features - done
serge-serge-1  |       -- The CXX compiler identification is unknown
serge-serge-1  |       CMake Error at CMakeLists.txt:4 (ENABLE_LANGUAGE):
serge-serge-1  |         No CMAKE_CXX_COMPILER could be found.
serge-serge-1  |       
serge-serge-1  |         Tell CMake where to find the compiler by setting either the environment
serge-serge-1  |         variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
serge-serge-1  |         to the compiler, or to the compiler name if it is in the PATH.
serge-serge-1  |       
serge-serge-1  |       
serge-serge-1  |       -- Configuring incomplete, errors occurred!
serge-serge-1  |       --
serge-serge-1  |       -------
serge-serge-1  |       ------------
serge-serge-1  |       -----------------
serge-serge-1  |       ----------------------
serge-serge-1  |       ---------------------------
serge-serge-1  |       --------------------------------
serge-serge-1  |       -- Trying 'Ninja' generator - failure
serge-serge-1  |       --------------------------------------------------------------------------------
serge-serge-1  |       
serge-serge-1  |       
serge-serge-1  |       
serge-serge-1  |       --------------------------------------------------------------------------------
serge-serge-1  |       -- Trying 'Unix Makefiles' generator
serge-serge-1  |       --------------------------------
serge-serge-1  |       ---------------------------
serge-serge-1  |       ----------------------
serge-serge-1  |       -----------------
serge-serge-1  |       ------------
serge-serge-1  |       -------
serge-serge-1  |       --
serge-serge-1  |       Not searching for unused variables given on the command line.
serge-serge-1  |       -- The C compiler identification is GNU 10.2.1
serge-serge-1  |       -- Detecting C compiler ABI info
serge-serge-1  |       -- Detecting C compiler ABI info - done
serge-serge-1  |       -- Check for working C compiler: /usr/bin/cc - skipped
serge-serge-1  |       -- Detecting C compile features
serge-serge-1  |       -- Detecting C compile features - done
serge-serge-1  |       -- The CXX compiler identification is unknown
serge-serge-1  |       CMake Error at CMakeLists.txt:4 (ENABLE_LANGUAGE):
serge-serge-1  |         No CMAKE_CXX_COMPILER could be found.
serge-serge-1  |       
serge-serge-1  |         Tell CMake where to find the compiler by setting either the environment
serge-serge-1  |         variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
serge-serge-1  |         to the compiler, or to the compiler name if it is in the PATH.
serge-serge-1  |       
serge-serge-1  |       
serge-serge-1  |       -- Configuring incomplete, errors occurred!
serge-serge-1  |       --
serge-serge-1  |       -------
serge-serge-1  |       ------------
serge-serge-1  |       -----------------
serge-serge-1  |       ----------------------
serge-serge-1  |       ---------------------------
serge-serge-1  |       --------------------------------
serge-serge-1  |       -- Trying 'Unix Makefiles' generator - failure
serge-serge-1  |       --------------------------------------------------------------------------------
serge-serge-1  |       
serge-serge-1  |                       ********************************************************************************
serge-serge-1  |                       scikit-build could not get a working generator for your system. Aborting build.
serge-serge-1  |       
serge-serge-1  |                       Building Linux wheels for Python 3.9 requires a compiler (e.g gcc).
serge-serge-1  |                 It can be installed using debian package manager:
serge-serge-1  |       
serge-serge-1  |       sudo apt-get install build-essential
serge-serge-1  |       
serge-serge-1  |                 To build compliant wheels, consider using the manylinux system described in PEP-513.
serge-serge-1  |                 Get it with "dockcross/manylinux-x64" docker image:
serge-serge-1  |       
serge-serge-1  |                   https://github.com/dockcross/dockcross#readme
serge-serge-1  |       
serge-serge-1  |                 For more details, please refer to scikit-build documentation:
serge-serge-1  |       
serge-serge-1  |                   http://scikit-build.readthedocs.io/en/latest/generators.html#linux
serge-serge-1  |       
serge-serge-1  |                       ********************************************************************************
serge-serge-1  |       [end of output]
```

Ok, I've added `build-essential` to the apt install, and that compiled this dependency, however it soon after failed on python `match` statement. 
```
serge-serge-1  |   File "/usr/src/app/api/src/serge/main.py", line 10, in <module>
serge-serge-1  |     from serge.routers.chat import chat_router
serge-serge-1  |   File "/usr/local/lib/python3.9/site-packages/serge/routers/__init__.py", line 1, in <module>
serge-serge-1  |     from .chat import chat_router
serge-serge-1  |   File "/usr/local/lib/python3.9/site-packages/serge/routers/chat.py", line 6, in <module>
serge-serge-1  |     from serge.utils.stream import get_prompt
serge-serge-1  |   File "/usr/local/lib/python3.9/site-packages/serge/utils/stream.py", line 61
serge-serge-1  |     match message.type:
serge-serge-1  |           ^
serge-serge-1  | SyntaxError: invalid syntax
serge-serge-1 exited with code 0
```

Was this commit intended ?

Reply:
:scream: Damn good catch. Of course it breaks, we switched to python 3.9 but the match statement is a python 3.10 addition. Absolutely my bad I should have caught it in testing. Will fix ASAP

Reply:
FYI, just checked out the commit before this merge (`git checkout 95ed087435095fdffddb806aa930182af7c344f6`) and it worked. 

Also, will be good to fix the `build-essential` missing dependency. 

Thanks :)

Reply:
@Deishelon Yes that's a bug. Which `arch` are you using that requires building wheels?